red_queen/cli.py

```diff
@@ -0,0 +1,185 @@
+# ------------------------------------------------------------------------------
+"""Shebang is used to make this code an python executable"""
+#!/usr/bin/env python3
+import os
+import platform
+import subprocess
+import configparser
+import click
+
+
+def benchmark_retrieval():
+    benchmark_dict = {}
+    type_list = []
+    list_of_benchmarks = []
+    dir_path = "red_queen/games/"
+    for entry in os.scandir(dir_path):
+        if entry.is_dir():
+            benchmark_dict[entry.name] = []
+            sub_dict = {}
+            for sub in os.scandir(f"{dir_path}{entry.name}"):
+                if not sub.name.startswith("_") and sub.name.endswith(".py") and sub.is_file():
+                    sub_dict[sub.name] = sub.path
+            benchmark_dict[entry.name] = sub_dict
+
+    type_list = list(benchmark_dict.keys())
+    for benchmark_pairs in benchmark_dict.values():
+        for keys in benchmark_pairs.keys():
+            list_of_benchmarks.append(keys)
+
+    return benchmark_dict, type_list, list_of_benchmarks
+
+
+def complier_retrieval():
+    list_of_compliers = []
+    config = configparser.ConfigParser()
+    config.read("pytest.ini")
+    for complier in config["pytest"]["markers"].split("\n"):
+        if complier != "":
+            list_of_compliers.append(complier)
+    return list_of_compliers
+
+
+def result_retrieval():
+    results = {}
+    dir_path = "results"
+    for entry in os.scandir(dir_path):
+        if entry.is_file():
+            filename = entry.name
+            result_count = int(filename.split("_")[0])
+            results[result_count] = {entry.name: entry.path}
+    return results
+
+
+def run_benchmarks(pytest_paths: str, m_tag: str, compiler: str):
+    command_list = ["pytest"]
+    compiler_command = [m_tag, compiler, "--store"]
+    if platform.system() == "Windows":
+        command_list.insert(0, "-m")
+        command_list.insert(0, "python")
+        command_list.append("-s")
+        for _, string in enumerate(pytest_paths):
+            command_list.append(string)
+        for _, string in enumerate(compiler_command):
+            command_list.append(string)
+        subprocess.run(
+            command_list,
+            check=True,
+        )
+    else:
+        for _, string in enumerate(pytest_paths):
+            command_list.append(string)
+        for _, string in enumerate(compiler_command):
+            command_list.append(string)
+        subprocess.run(
+            command_list,
+            check=True,
+        )
+
+
+def show_result():
+    results_dict = result_retrieval()
+    result_num = max(results_dict.keys())
+    result_path = tuple(results_dict[result_num].values())[0]
+    command_list = ["python3", "-m", "report.console_tables", "--storage"]
+    command_list.append(str(result_path))
+    subprocess.run(
+        command_list,
+        check=True,
+    )
+    click.echo("To view the table again:")
+    click.echo(" ".join(command_list))
```

Comment:
Ideally we should be able to call this directly via python. It looks like I forgot to move the `report` directory into `red_queen` as part of #21 which precludes doing this now. So this is fine for now. But after this merges we should move the report stuff into the red queen package and then basically make this function something like:

```
from red_queen.report import table_visualizer

table_visualizer(result_path)
```

Reply:
If anything, once we get the CLI merged, open that as an issue. I'll try my best to address it in a timely manor.